### PR TITLE
[Do Not Merge Yet] Add gitlab builder support

### PR DESCRIPTION
**Pull request by venezia**
This PR will add a gitlab-ci support.

It changes the test runner slightly to no longer pull credentials from an S3 bucket (which access is granted through an IAM role on nodes running the job) in favor of environmental variables in the job itself.

It can also remove images generated on the local container registry if you provide valid credentials

Please see the instructions in the .gitlab-ci.yml file

I marked this do not merge as i commented out the `aws s3` commands, which is still needed until fail-fast integration is complete, but i guess i could leave them in there with them failing, but unsure.

LMK

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    DOCKER_DRIVER: overlay2
+    DOCKER_REGISTRY: $CI_REGISTRY
+    DOCKER_PROJECT: $CI_PROJECT_PATH
+    DOCKER_TAG: branch-$CI_COMMIT_REF_NAME-$CI_PIPELINE_ID
+  only:
+    - branches
+    - master
+  when: always
+  allow_failure: true
+  image: registry.cnct.io/venezia/docker-registry-curl:v0.1.0
```
Comment (leahnp) [inline .gitlab-ci.yml]: What does this do?

Comment (venezia) [reply .gitlab-ci.yml]: Can you be a bit more explicit in what you want to know?

Comment (samba) [reply .gitlab-ci.yml]: @leahnp the image he's referencing is hosted by GitLab's internal Docker registry. The purpose of that image is to facilitate cleanup of the registry's images, after they're validated (especially failing). 

Notably our kraken images are quite large, so an abundance of them becomes quite expensive to store. @venezia took great pains to incorporate a cleanup phase for that project's CI.

**Comment by joejulian** [comment]
Closing and re-opening to allow failfast-ci to pick this up.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,199 @@
+#
+# Kraken Build File for GitLab
+# This pipeline requires 11 variables to be set in the CI/CD Settings
```
Comment (philoserf) [inline .gitlab-ci.yml]: I like this comment block!

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  - docker push $IMAGE_URI
+
+.aws-test: &aws_test_definition
+  stage: test
+  variables:
+    JOB_BASE_NAME: ${CI_PROJECT_NAME}
+    BUILD_ID: $CI_PIPELINE_ID
+  image: $IMAGE_URI
+  script:
+  - mkdir -p /root/.ssh /root/.aws
```
Comment (philoserf) [inline .gitlab-ci.yml]: I prefer the script be a script or a make task. That way a human can easily duplicate locally what the CI systems is doing. I'll prepare a short presentation on this for the wider team.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    - docker info
+    - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
+  script:
+  - docker build -t $IMAGE_URI .
+  - docker push $IMAGE_URI
+
+.aws-test: &aws_test_definition
+  stage: test
+  variables:
+    JOB_BASE_NAME: ${CI_PROJECT_NAME}
```
Comment (philoserf) [inline .gitlab-ci.yml]: `CI_JOB_ID` is unique. While I see in Jenkins we built names from multiple variables I do not think we need to keep doing that here.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (philoserf) [inline .gitignore]: While we commonly see this I think editor and os git ignore directives belong to `$HOME/.gitignore_global`

GitHub illustrates this by what they keep in the base and what they put in the Global folder here: https://github.com/github/gitignore

**Comment by coffeepac** [comment]
this work was accomplished in [another PR](#158).  I am closing

**Pull request closed without merging**